src/concrete_intersection.jl

```diff
@@ -656,3 +656,27 @@ end
 function intersection(L1::LinearMap{N}, L2::LinearMap{N}) where {N}
     return intersection(linear_map(L1.M, L1.X), linear_map(L2.M, L2.X))
 end
+
+"""
+intersection(ca1::CartesianProductArray{N, S}, L2::CartesianProductArray{N, S}, nonzero_blocks::Vector{Int})
+              ) where {N<:Real, S<:LazySet{N}}
+
+Return the intersection of two CartesianProductArray's only for necessary blocks.
+
+### Input
+
+ - `ca1` -- Cartesian Product Array of convex sets
+ - `ca2` -- Cartesian Product Array of LazySets (it might by unbounded sets)
+ - `nonzero_blocks` -- Required blocks to intersect
+
+### Output
+
+The Cartesian Product Array obtained by the intersection nonzero blocks of `ca1` and `ca2`.
+"""
+function intersection(ca1::CartesianProductArray{N, S}, L2::CartesianProductArray{N, S}, nonzero_blocks::Vector{Int})
```

Comment:
`nonzero_blocks` should be an *optional* argument. The code should run without having to specify this. You can write a function to compute the default value.

`L2` should be `ca2` I guess.